packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -684,6 +728,24 @@ export abstract class BaseService extends Resource
     };
   }
 
+  /**
+   *   Add an alarm based on ECS metric
+  */
+  public createEcsMetricAlarm(metric: EcsMetric, ecsMetricAlarmProps?: EcsMetricAlarmProps): cloudwatch.Alarm {
+    // Throw an error if service connect is not configured
+    if (!this._serviceConnectConfig) {
+      throw new Error('Service connect must be enabled to set service connect metric alarms.');
+    }
+    const ecsMetric = this.metric(metric, ecsMetricAlarmProps?.metricProps);
+    const metricAlarm = new cloudwatch.Alarm(this, `${metric}Alarm`, ecsMetricAlarmProps?.useAsDeploymentAlarm && ecsMetricAlarmProps?.alarmProps ? ecsMetricAlarmProps.alarmProps : {
+      metric: ecsMetric as IMetric,
+      threshold: ecsMetricAlarmProps?.alarmProps?.threshold || 1,
+      evaluationPeriods: ecsMetricAlarmProps?.alarmProps?.evaluationPeriods || 1,
+    });
+    this.deploymentAlarms?.alarmNames.push(metricAlarm.alarmName);
```

Comment:
What happens if deploymentAlarms doesn't exist yet? We should either validate that there is a deployment alarm configuration, or set default values for deploymentalarm configuration at the beginning of this method (enabled, rollback: true)